src/moonlink_connectors/src/rest_ingest/decimal_utils.rs

```diff
@@ -0,0 +1,190 @@
+use bigdecimal::BigDecimal;
+use moonlink::row::RowValue;
+use std::convert::TryInto;
+use std::str::FromStr;
+use thiserror::Error;
+
+#[derive(Debug, Error, PartialEq)]
+pub enum DecimalConversionError {
+    #[error("Decimal normalization precision failed (value: {value}, parsed precision: {parsed_precision})")]
+    NormalizationPrecision {
+        value: String,
+        parsed_precision: usize,
+    },
+    #[error("Decimal normalization scale failed (value: {value}, parsed scale: {parsed_scale})")]
+    NormalizationScale { value: String, parsed_scale: i64 },
+    #[error("Decimal precision exceeds the specified precision (value: {value}, expected ≤ {expected_precision}, actual {actual_precision})")]
+    PrecisionMismatch {
+        value: String,
+        expected_precision: u8,
+        actual_precision: u8,
+    },
+    #[error("Decimal scale exceeds the specified scale (value: {value}, expected ≤ {expected_scale}, actual {actual_scale})")]
+    ScaleMismatch {
+        value: String,
+        expected_scale: i8,
+        actual_scale: i8,
+    },
+    #[error("Decimal value is invalid: {value})")]
+    InvalidValue { value: String },
+    #[error("Decimal mantissa overflow: {mantissa}, error: {error}")]
+    Overflow { mantissa: String, error: String },
```

Comment:
nit
```suggestion
    Overflow { mantissa: String, err_msg: String },
```
I see commonly people use `err` / `error` for `Error` type, while `err_msg` / `error_message` for its string format